### PR TITLE
Simplify keyboard layout handling, rely on localed more (Fedora 39)

### PR DESCRIPTION
**Pull request by AdamWill**
There are a couple of specific bugs in the current code:

* https://bugzilla.redhat.com/show_bug.cgi?id=2238854
* https://bugzilla.redhat.com/show_bug.cgi?id=2239213

which could both be solved by just...trying less hard. Let's not try to set the console layout directly on the old set_x_keyboard_defaults path, and let's not have special code to set the console layout when reading configs from the live image, let's just read in the X layouts from the live config then rely on the existing code to ask localed to pick a console layout for us.

This also sets the keyboard model to "pc105" when requesting a console layout from localed. This is because localed only applies match "bonuses" for matching variants and options if the model matches. We need the variant "bonus" match to get the right result for Bulgarian, at least. Almost every line in kbd-model-map has the model set to "pc105", so let's just go with that. The only line with a different model *and* a variant is es-dvorak, and there is an xkb-converted layout for that, so it shouldn't make any difference there.

With current stable systemd this will give us wrong results in several cases (e.g. Russian installs will get US console layout). With current git HEAD systemd - after
https://github.com/systemd/systemd/pull/29215 - I believe this gives the same or better results in all cases on both new and old workflows (I have tested at least Russian and several Bulgarian paths). With
https://github.com/systemd/systemd/pull/29236 on top, we also get the correct result for Bulgarian with no variant, "bg_bds-utf8". Without that PR but with this change, we get the incorrect "bg_pho-utf8" in that case. Without this change, we set the console layout to "bg", which does not exist and causes errors on system startup and fallback to the "us" layout.

Resolves: rhbz#2238854
Resolves: rhbz#2239213

### CODE REVIEW

**Comment by AdamWill** [comment]
Note, we shouldn't merge this till a systemd with at least my first two relevant changes is stable for F39. zbyszek is working on an update right now which will have them.

**Review by jkonecny12 — APPROVED** [review]

**Comment by jkonecny12** [comment]
We are waiting until everything is landed into Fedora. @AdamWill will ping us here when we can merge this.

**Comment by AdamWill** [comment]
https://bodhi.fedoraproject.org/updates/FEDORA-2023-3355cdb0a8 is the update, I will try and get it pushed.

**Comment by AdamWill** [comment]
sigh, I didn't get around to that and now we're frozen. I've marked the systemd update as related to the blocker bug, and also proposed another bug it fixes as an FE. That should let us push it.

**Comment by AdamWill** [comment]
OK, the systemd update is pushed stable, so it should be fine to merge this now.

**Comment by M4rtinK** [comment]
> OK, the systemd update is pushed stable, so it should be fine to merge this now.

OK then, merging now & doing a build for F39 & Rawhide with this.

**Comment by M4rtinK** [comment]
> OK then, merging now & doing a build for F39 & Rawhide with this.

Correction - looks like this has been already released on Rawhide. :)

**Pull request merged**
